cli/src/opts/mod.rs

```diff
@@ -72,6 +72,7 @@ pub struct EthereumOpts {
 
     #[clap(long, env = "ETHERSCAN_API_KEY")]
     pub etherscan_api_key: Option<String>,
+
```

Comment:
i originally planned to support the `legacy` flag but it seems like a ton of plumbing to get it to `build_tx`. not sure if there is a better way

Reply:
yeah the `build_tx` has become a bit too verbose.

let's stick with `chain` only, because then we can't run into scenarios where legacy flag is set but chain is set to eip1559, instead this should then be added to the `is_legacy` function directly.